File: src/main/java/ro/ubb/domain/Proposal.java
package ro.ubb.domain;

import lombok.*;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "proposal")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@EqualsAndHashCode
@ToString
public class Proposal {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "proposal_id")
  private Integer proposalID;

  @OneToMany(mappedBy = "proposal", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
  Set<Review> reviews = new HashSet<>();

  @OneToMany(mappedBy = "proposal", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
  Set<Author> authors = new HashSet<>();

  @OneToMany(mappedBy = "proposal", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
  Set<Analysis> analyses = new HashSet<>();

  @OneToMany(mappedBy = "proposal", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
  Set<ProposalList> proposalListsForSections = new HashSet<>();

  @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
  @JoinColumn(name = "user_info_id")
  private UserInfo userInfo;

  @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
  @JoinColumn(name = "meta_info_id")
  private MetaInformation metaInformation;

  @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
  @JoinColumn(name = "abstract_id")
  private Abstract anAbstract;

  @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
  @JoinColumn(name = "paper_id")
  private Paper paper;
}

File: client/CMS/src/app/shared/models/Proposal.ts
export class Proposal {
   proposalID: number;
   userInfoID: number;
   paperID: number;
   metaInfoID: number;
   abstractID: number;

  constructor() {
    this.proposalID = 0;
    this.userInfoID = 0;
    this.paperID = 0;
    this.metaInfoID = 0;
    this.abstractID = 0;
  }

  setAttributes(proposalID: number, userInfoID: number, paperID: number, metaInfoID: number, abstractID: number) {
    this.proposalID = proposalID;
    this.userInfoID = userInfoID;
    this.paperID = paperID;
    this.metaInfoID = metaInfoID;
    this.abstractID = abstractID;
  }

  public getProposalID(): number {
    return this.proposalID;
  }
}

export class Proposals{
  proposalDtoList: Array<Proposal>;
}
